packages/core-api/src/extensions/extensions.tsx

```diff
@@ -61,15 +61,13 @@ export function createRoutableExtension<
 }
 
 export function createComponentExtension<
-  T extends (props: any) => JSX.Element
+  T extends ComponentType<any>
 >(options: { component: ComponentLoader<T> }): Extension<T> {
   const { component } = options;
   return createReactExtension({ component });
 }
 
-export function createReactExtension<
-  T extends (props: any) => JSX.Element
->(options: {
+export function createReactExtension<T extends ComponentType<any>>(options: {
```

Comment:
We want to avoid `ComponentType` here as it doesn't properly let us convey the `children` props. `ComponentType` inserts it as an optional prop whether the inner component accepts it or not, making it impossible to make the usage of children type safe.

I think what we want to do instead is to simply allow `null` as a return value on top of `JSX.Element`, perhaps even just switching it out to `ReactNode`, but I think we can start with `null`.

This all should've been in a comment here, so happy if you want to add that too :grin: